packages/zk-transfer-web/src/components/Token.tsx

```diff
@@ -1,31 +1,55 @@
 // @ts-ignore
-import React, { useState, useEffect, useRef } from "react";
+import React, { useState, useCallback, useEffect, useRef } from "react";
 import arrowLeft from "../assets/arrow-left.svg";
+// @ts-ignore
+import { note } from "aztec.js"; 
+import { NoteList, NoteValue } from "../libs/types";
+import { getSum } from "../util";
 
 type TokenPropType = {
-  firstValue: number;
-  secondValue: number;
+  name: string;
+  firstValue?: number;
+  secondValue?: number;
   children: any;
   canEdit?: boolean;
+  notes?: NoteList;
   onSend: Function;
   allowEdit?: Function;
 };
 
+const fetchValue = (notes: NoteList) => {
+  const getNoteValue = function (e: any) {
+    return note.fromViewKey(e.get("viewingKey"))
+  };
+  const pendingValues: NoteValue[] = notes.map(getNoteValue);
+  return Promise.all(Array.from(pendingValues));
+}
+
 const TokenInput = (props: TokenPropType) => {
-  const editMode = props.canEdit;
-  const [digit, setDigit] = useState(props.firstValue);
   const input = useRef(null);
+  const editMode = props.canEdit;
+  const [digit, setDigit] = useState(0);
```

Comment:
what is `digit`?
Why save the `fetched` state?
I thought that React suspense and fetchers handled caching values internally.

Reply:
So, i wrapped the logic in a Custom Hook insted of having to use the React suspense (since it's still experimental).